### PR TITLE
Add callAs parameter to StackSet Resource

### PR DESCRIPTION
**Pull request by aradven**
*Issue #, if available:*

*Description of changes:*
Adds the callAs parameter to the StackSet resource, enabling use of delegated administration

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/ReadHandler.java [hunk 1]
```diff
@@ -23,8 +23,9 @@
         return ProgressEvent.<ResourceModel, CallbackContext>builder()
                 .resourceModel(ResourceModelBuilder.builder()
                         .proxyClient(proxyClient)
-                        .stackSet(describeStackSet(proxyClient, model.getStackSetId()))
-                        .build().buildModel())
+                        .stackSet(describeStackSet(proxyClient, model.getStackSetId(), model.getCallAs()))
+                        .build()
+                        .buildModel(model.getCallAs()))
```
Comment (xiwhuang) [inline aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/ReadHandler.java:28]: I am afraid, we might not be able to `getCallAs` in read handler as it is not one of the identifiers.

File: aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ResourceModelBuilder.java [hunk 2]
```diff
@@ -32,12 +32,15 @@
...
+    public ResourceModel buildModel() {
+        return buildModel(null);
+    }
     /**
      * Returns the model we construct from StackSet service client using PrimaryIdentifier StackSetId
      *
      * @return {@link ResourceModel}
      */
-    public ResourceModel buildModel() {
+    public ResourceModel buildModel(final String callAs) {
```
Comment (xiwhuang) [inline aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ResourceModelBuilder.java:43]: Nit: We can actually add `CallAs` as a member variable to be consistent.

Comment (aradven) [reply aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ResourceModelBuilder.java:43]: I understand why, but I think I'd rather not here. CallAs is not really part of the actual model, it's part of the request, and I think we should reflect that where possible

**Review by xiwhuang — APPROVED** [review]
LGTM, just to make sure it works for resource import as well

File: aws-cloudformation-stackset/src/test/java/software/amazon/cloudformation/stackset/CreateHandlerTest.java [hunk 3]
```diff
@@ -99,6 +102,45 @@ public void handleRequest_ServiceManagedSS_SimpleSuccess() {
         verify(proxyClient.client()).describeStackSetOperation(any(DescribeStackSetOperationRequest.class));
     }
 
+    @Test
+    public void handleRequest_ServiceManagedSS_WithCallAs_SimpleSuccess() {
```
Comment (xalle-y) [inline aws-cloudformation-stackset/src/test/java/software/amazon/cloudformation/stackset/CreateHandlerTest.java:106]: Any reason not to include tests for SelfManaged with CallAs?

Comment (aradven) [reply aws-cloudformation-stackset/src/test/java/software/amazon/cloudformation/stackset/CreateHandlerTest.java:106]: Mostly cause there's no functional difference within the handler's themselves. It's just another parameter that's passed through to the SDK, we don't do logic with it here.

Comment (xalle-y) [reply aws-cloudformation-stackset/src/test/java/software/amazon/cloudformation/stackset/CreateHandlerTest.java:106]: We should not be assuming this type of behaviour when deciding what should or should not be tested. The main addition here is "CallAs for Self and Service-managed stack sets" so expected behaviour for those cases should be tested. Even if it's a no-op just having the test there is a better way to solidify that expectation.

Comment (aradven) [reply aws-cloudformation-stackset/src/test/java/software/amazon/cloudformation/stackset/CreateHandlerTest.java:106]: Fair enough, I'll add some

File: aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ResourceModelBuilder.java (hunk 2 above)
Comment (aradven) [reply aws-cloudformation-stackset/src/main/java/software/amazon/cloudformation/stackset/util/ResourceModelBuilder.java:43]: Realized that import would actually probably rely on this existing, so adding it to the model

**Review by xalle-y — APPROVED** [review]

**Pull request merged**